mitosheet/mitosheet/api/get_imported_files_and_dataframes.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+import json
+import os
+from typing import Any, Dict
+
+import pandas as pd
+from mitosheet.types import StepsManagerType
+
+
+def get_imported_files_and_dataframes(params: Dict[str, Any], steps_manager: StepsManagerType) -> str:
+    """
+    Returns a list all the imported files and dataframes, and their import params.
+    Note: we break each imported file into its own object so that if a user imported multiple dataframes
+    through the same dataframe_import step, they can replace one with a csv and the other with a new df, for example. 
+
+    [
+        {
+            step_id: str,
+            type: 'csv'
+            import_params: {
+                file_names: List[str] <- Note: This should only have one entry, 
+                encoding: str | undefined, 
+                delimeters: str | undefined,
+                error_bad_lines: boolean | undefined
+            } 
+        } |
+        {
+            step_id: str,
+            type: 'excel'
+            import_params: {
+                file_name: str 
+                sheet_names: List[str] <- Note: This should only have one entry
+                has_headers: bool
+                skiprows: int 
+            }
+        } |
+        {
+            step_id: str
+            type: 'df'
+            df_names: List[str] <- Note: This should only have one entry
+        }
+	]
+	
+    """
+    imported_files_and_dataframes = []
+    for step in steps_manager.steps_including_skipped:
+        step_params_copy = step.params.copy()
```

Comment:
This looks a bit scary to me, especially since we're deleting from it later. I wonder if making it a copy before returning it is safer, just so if things get moved around this line doesn't get lost.